lib/rack/request.rb

```diff
@@ -79,6 +79,88 @@ def delete_param(k)
       v
     end
 
+    def headers
+      @headers ||= Headers.new(@env)
+    end
+
+    class Headers
+      def initialize(env)
+        @env = env
+      end
+
+      def [](k)
+        @env[header_to_env_key(k)]
+      end
+
+      def []=(k, v)
+        @env[header_to_env_key(k)] = v
+      end
+
+      def add(k, v)
```

Comment:
Why do you `add` instead `set` name method @jeremyevans ?

Reply:
Because this appends and does not overwrite.